custom_components/tahoma/alarm_control_panel.py

```diff
@@ -0,0 +1,164 @@
+
+            if state == STATE_OFF:
+                alarm_state = STATE_ALARM_DISARMED
+            elif state == STATE_PARTIAL_1:
+                alarm_state = STATE_ALARM_ARMED_HOME
+            elif state == STATE_PARTIAL_2:
+                alarm_state = STATE_ALARM_ARMED_NIGHT
+            elif state == STATE_TOTAL:
+                alarm_state = STATE_ALARM_ARMED_AWAY
+
+        if self.has_state(CORE_INTRUSION_STATE, INTERNAL_INTRUSION_DETECTED_STATE):
+            state = self.select_state(
+                CORE_INTRUSION_STATE, INTERNAL_INTRUSION_DETECTED_STATE
+            )
+
+            if state == STATE_DETECTED:
+                alarm_state = STATE_ALARM_TRIGGERED
+            elif state == STATE_PENDING:
+                alarm_state = STATE_ALARM_PENDING
+
+        return alarm_state
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        supported_features = 0
+
+        if self.has_command(COMMAND_ARM, COMMAND_ALARM_ON):
+            supported_features |= SUPPORT_ALARM_ARM_AWAY
+
+        if self.has_command(COMMAND_ALARM_PARTIAL_1):
+            supported_features |= SUPPORT_ALARM_ARM_HOME
+
+        if self.has_command(COMMAND_PARTIAL, COMMAND_ALARM_PARTIAL_2):
+            supported_features |= SUPPORT_ALARM_ARM_NIGHT
+
+        if self.has_command(COMMAND_SET_ALARM_STATUS):
+            supported_features |= SUPPORT_ALARM_TRIGGER
+            supported_features |= SUPPORT_ALARM_ARM_CUSTOM_BYPASS
+
+        return supported_features
+
+    async def async_alarm_disarm(self, code=None):
+        """Send disarm command."""
+        await self.async_execute_command(
+            self.select_command(COMMAND_DISARM, COMMAND_ALARM_OFF)
+        )
+
+    async def async_alarm_arm_home(self, code=None):
+        """Send arm home command."""
+        await self.async_execute_command(self.select_command(COMMAND_ALARM_PARTIAL_1))
+
+    async def async_alarm_arm_night(self, code=None):
+        """Send arm night command."""
+        await self.async_execute_command(
+            self.select_command(COMMAND_PARTIAL, COMMAND_ALARM_PARTIAL_2)
+        )
+
+    async def async_alarm_arm_away(self, code=None):
+        """Send arm away command."""
+        await self.async_execute_command(
+            self.select_command(COMMAND_ARM, COMMAND_ALARM_ON)
+        )
+
+    async def async_alarm_trigger(self, code=None) -> None:
+        """Send alarm trigger command."""
+        await self.async_execute_command(COMMAND_SET_ALARM_STATUS, STATE_DETECTED)
+
+    async def async_alarm_arm_custom_bypass(self, code=None) -> None:
+        """Send arm custom bypass command."""
+        await self.async_execute_command(COMMAND_SET_ALARM_STATUS, STATE_UNDETECTED)
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        # if self.device.widget == "TSKAlarmController":
```

Comment:
Not sure if this is the best way, but I could not figure out yet which state shows that someone is actually using the controller...